### PR TITLE
Mac bug fixes

### PR DESCRIPTION
**Pull request by pemba1s1**
**Issue** #134 
### Fixes:
- Mac Icon
- LoadingWindow Bug
- Button Color
- Crashing due to thread not closed properly in settings
- Need microphone permission

## Summary by Sourcery

Fix multiple bugs related to the Mac client, including icon display, button color, and thread management issues. Enhance the build scripts for better readability and add microphone permission for macOS.

Bug Fixes:
- Fix the Mac icon display issue by using a new utility function to set the logo.
- Resolve the LoadingWindow bug by adjusting the window management logic.
- Correct button color inconsistencies by adding highlight background configurations.
- Fix application crash by ensuring proper closure of threads in the settings.
- Add microphone permission description for macOS builds to prevent permission-related issues.

Enhancements:
- Refactor icon setting logic into a new utility function for better code reuse and maintainability.

Build:
- Update build scripts to improve readability and maintainability by using multi-line commands.

### CODE REVIEW

File: src/FreeScribe.client/client.py [hunk 1]
```diff
@@ -140,19 +143,19 @@ def toggle_pause():
...
         elif current_view == "minimal":
-            pause_button.config(text="▶️", bg="red")
+            pause_button.config(text="▶️", bg="red", highlightbackground="red")
     else:
         if current_view == "full":
-            pause_button.config(text="Pause", bg=DEFAULT_BUTTON_COLOUR)
+            pause_button.config(text="Pause", bg=DEFAULT_BUTTON_COLOUR, highlightbackground=DEFAULT_BUTTON_COLOUR)
         elif current_view == "minimal":
-            pause_button.config(text="⏸️", bg=DEFAULT_BUTTON_COLOUR)
+            pause_button.config(text="⏸️", bg=DEFAULT_BUTTON_COLOUR, highlightbackground=DEFAULT_BUTTON_COLOUR)
```
[the remark this answers is not in the document]
Comment (pemba1s1) [reply src/FreeScribe.client/client.py:158]: have one more if block above it. changing the bottom one to elif will make the code harder to read/understand

File: src/FreeScribe.client/UI/LoadingWindow.py [hunk 2]
```diff
@@ -130,12 +127,8 @@ def destroy(self):
...
         """
         if self.popup:
-            # Enable the parent window
-            if self.parent:
-                self.parent.wm_attributes('-disabled', False)
-            
             if self.progress.winfo_exists():
                 self.progress.stop()
 
             if self.popup.winfo_exists():
```
[the remark this answers is not in the document]
Comment (pemba1s1) [reply src/FreeScribe.client/UI/LoadingWindow.py:134]: done

File: src/FreeScribe.client/UI/Widgets/AudioMeter.py [hunk 3]
```diff
@@ -76,9 +76,15 @@ def cleanup(self, event=None):
             return
 
         self.destroyed = True
```
[the remark this answers is not in the document]
Comment (pemba1s1) [reply src/FreeScribe.client/UI/Widgets/AudioMeter.py:80]: used thread event instead to monitor

**Review by yingbull — CHANGES_REQUESTED** [review]
Looks like some Sourcery feedback not resolved? Tag me if you want to discuss.

File: src/FreeScribe.client/utils/file_utils.py [hunk 4]
```diff
@@ -15,7 +14,7 @@ def get_file_path(*file_names: str) -> str:
     base = sys._MEIPASS if hasattr(sys, '_MEIPASS') else os.path.abspath('.')
     return os.path.join(base, *file_names)
 
-def get_resource_path(filename: str) -> str:
+def get_resource_path(*filename: str, shared: bool = False) -> str:
     """
```
[the remark this answers is not in the document]
Comment (pemba1s1) [reply src/FreeScribe.client/utils/file_utils.py:18]: used guard condition

**Comment by pemba1s1** [comment]
@yingbull Resolved the feedback. Ready for review

File: src/FreeScribe.client/client.py [hunk 5]
```diff
@@ -460,6 +463,8 @@ def send_audio_to_server():
         # Display a message indicating that audio to text processing is in progress
         user_input.scrolled_text.insert(tk.END, "Audio to Text Processing...Please Wait")
         try:
+            if getattr(sys, 'frozen', False) and sys.platform == 'darwin':  # Check if running as a bundled app in macOS
```
[the remark this answers is not in the document]
Comment (pemba1s1) [reply src/FreeScribe.client/client.py:520]: not a duplicate code

**Review by yingbull — APPROVED** [review]

**Pull request merged**